src/dimension.c

```
@@ -756,9 +768,10 @@ dimension_update(FunctionCallInfo fcinfo,
 	if (NULL != interval)
 	{
 		Oid			intervaltype = get_fn_expr_argtype(fcinfo->flinfo, 1);
+		Oid			valuetype = OidIsValid(dim->time_func) ? dim->time_func_rettype : dim->fd.column_type;
```

This function also allows setting both an `interval` and `num_slices` for a dimension.

---

Added an assert to the tuple update function, which corresponds to the tuple insert case where we have the other `Assert`. Again, note that our schema also protects us against inserting inconsistent values.

While I am not against `Asserts`, we probably need to draw the line somewhere. Otherwise we could also assert that all other values make sense, are not `NULL`, `0`, etc. And should we do this in all functions that handle these values?

I think we need some guidelines for asserts. For instance, user-facing functions (like, e.g., `dimension_add`) should probably check parameter inconsistencies and return user-level errors. We can then double-check consistency with `Assert` when these values are persisted, like in `dimension_insert_relation` or `dimension_update_tuple`. 

Otherwise I think `Asserts` make sense in two cases:

1. For clarity/readability. You'd like to convey to someone reading the code that are "impossible" cases that are extra important.
2. Data integrity. You would rather fail hard then end up with corrupted data.

---

Asserts should not be used for data integrity because they will not run in production code.